mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/AnnotationMapper.java

```diff
@@ -84,4 +84,20 @@
      */
     void insertAnnotation(MailboxId mailboxId, MailboxAnnotation mailboxAnnotation);
 
+    /**
+     * Checking the current annotation of selected mailbox is existed on store or not.
```

Comment:
I guess you don't check the value but only if any entry has the same key already ?